ginac/numeric.cpp

```diff
@@ -3241,8 +3241,8 @@ long numeric::to_long() const {
                 mpz_init(bigint);
                 mpz_fdiv_q(bigint, mpq_numref(v._bigrat),
                                 mpq_denref(v._bigrat));
-                if (mpz_fits_sint_p(v._bigint)) {
-                        long n = mpz_get_si(bigint);
+                if (mpz_fits_slong_p(v._bigint)) {
+                        long n = (long int)mpz_get_si(bigint);
```

Comment:
why does one need an explicit cast here? The return type of `mpz_get_si` is `long int` anyway.